First attempt

areian opened the pull request.

areian committed:
Add dep files

areian committed:
Initial work on the RDB reader

areian committed:
Initial work on Field length parsing

Unfinished

areian committed:
Prepwork for stuff to come

kasperlewau commented on rdb.go:
```diff
@@ -0,0 +1,165 @@
+	}
+	switch b >> 6 {
+	case 0: // 6 bit integer
+		return []byte{b << 2 >> 2}, nil
```
Now it's getting really exciting! :smile:

kasperlewau commented on rdb.go:
```diff
@@ -0,0 +1,165 @@
+	if p, err := buffer.Peek(5); err != nil || !bytes.Equal([]byte("REDIS"), p) {
+		return nil, ErrFormat
+	}
+	buffer.Discard(5) // Skip peeked bytes
```
Turn into a reader, break if first read does not give you the expected output

Commit:
Read bytes rather than Peek/Discard

Commit:
Reuse buffer var

Commit:
Add tests to increase coverage

Commit:
Add more error handling for malformed/corrupted files

Commit:
readFieldLenght returns uint64, rather than []byte

Commit:
Parse length for strings

Commit:
Only parse versions we have tested and know to be good

Commit:
Read Metadata from Dump

Commit:
Better OpCode handling

Commit:
Finish SetDBNo

Commit:
Skip DB size info if pressent, when reading DB number

Commit:
Read Metadata as part of Reader initialization

Commit:
Rename test to allign with function name

Commit:
wip: Read Redis Entries

areian committed:
Refactor

Broke out String parsing into its own function
Rename: readFieldLenth -> readLenghtEncodedValue
Moved readLenghtEncodedValue to bottom, to group helper functions

Commit:
Move test for reading length encoded data to match placment in rdb.go

Commit:
Move tests for reading String encoding to dedicated test function

Commit:
Read methods now also return the raw bytes read to parse these values

This is done to make it easier to run restore commands

Commit:
Read List values

areian committed:
Add Set decoding

areian committed:
Linting

areian committed:
Use bit masks instead of bit-shifting
